python/ray/data/random_access_dataset.py

```diff
@@ -0,0 +1,265 @@
+import bisect
+import logging
+import random
+import time
+from collections import defaultdict
+from typing import List, Any, Generic, Optional, TYPE_CHECKING
+
+import ray
+from ray.types import ObjectRef
+from ray.data.block import T, BlockAccessor
+from ray.data.impl.remote_fn import cached_remote_fn
+
+if TYPE_CHECKING:
+    from ray.data import Dataset
+
+logger = logging.getLogger(__name__)
+
+
+class RandomAccessDataset(Generic[T]):
+    """A class that provides distributed, random access to a Dataset.
+
+    See: ``Dataset.to_random_access_dataset()``.
+    """
+
+    def __init__(
+        self,
+        dataset: "Dataset[T]",
+        key: str,
+        num_workers: int,
+    ):
+        """Construct a RandomAccessDataset (internal API).
+
+        The constructor is not part of the Datasets API. Use
```

Comment:
It's a bit difficult to parse this. Does it just mean the ctor is a private/internal API?

Reply:
Fixed.